plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/FacetInstallDelegateTest.java

```diff
@@ -0,0 +1,21 @@
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class FacetInstallDelegateTest {
+
+  private FacetInstallDelegate delegate = new FacetInstallDelegate();
+  
+  @Test
+  public void testMavenNature() throws CoreException {
+    IProject project = Mockito.mock(IProject.class);
+    Mockito.when(project.hasNature("org.eclipse.m2e.core.maven2Nature")).thenReturn(true);
+    Mockito.when(project.isAccessible()).thenReturn(true);
+    
+    delegate.execute(project, null, null, null);
```

Comment:
Ugly and not sure we should do it, but we could assert on project.getType() being called which is invoked during the JavaCore.create(...) method execution and this way we could verify if the body of the `if` statement is executed. Otherwise this test is not useful.